### DISCUSSION TITLE
Valheim server dead (Caught fatal signal)

### DISCUSSION THREAD

**Discussion by Ksando** [GameServerManagers/LinuxGSM#4088] [Q&A]
There is a dedicated server on Centos 8. Resources were gradually reduced to 2 cores and 4Gb of RAM, because we play together or three.
At first, there were no problems and the server worked without interruption for many days. Then there was the first fall and since then the falls have become more and more frequent. At the moment, the server could not work for an hour, until everything fell with that error.
In an adjacent topic (https://github.com/GameServerManagers/LinuxGSM/discussions/4058), with a similar problem, it was advised to remove the -crossplay launch mode and remove the libparty.so library. The -crosplay mode was already missing for me, and deleting the library did not help.
I am attaching the command to start the server and the log at the time of the crash:

> ./valheim_server.x86_64 -nographics -batchmode -name "XXXXX" -port 2456 -world "XXXXX" -password "XXXXX" -public "1" -savedir "/opt/valheim/world/" -logFile "/opt/valheim/logs/log_current.log" -saveinterval "600"

> 01/20/2023` 01:19:40:  Connections 1 ZDOS:453767  sent:0 recv:554
Unloading 0 Unused Serialized files (Serialized files now loaded: 0)
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_lowlevel.cpp (1873) : Assertion Failed: SteamnetworkingSockets service thread waited 147ms for lock!  This directly adds to network latency!  It could be a bug, but it's usually caused by general performance problem such as thread starvation or a debug output handler taking too long.
Could not allocate 67108864 bytes
Caught fatal signal - signo:6 code:-6 errno:0 addr:0x3e80001a56f
Obtained 22 stack frames.
#0  0x007fd7a3fdfcf0 in funlockfile
#1  0x007fd7a3c56acf in gsignal
#2  0x007fd7a3c29ea5 in abort
#3  0x007fd7a329cae8 in mono_dl_fallback_unregister
#4  0x007fd7a32ace94 in monoeg_g_logv
#5  0x007fd7a32acf27 in monoeg_g_log
#6  0x007fd7a32acb3a in monoeg_realloc
#7  0x007fd7a32afb54 in monoeg_g_string_erase
#8  0x007fd7a32afb92 in monoeg_g_ptr_array_sized_new
#9  0x007fd7a3292b3e in mono_monitor_enter_v4
#10 0x007fd7a3292d04 in mono_unity_liveness_allocate_struct
#11 0x007fd7a3293a2f in mono_unity_liveness_calculation_begin
#12 0x007fd7a54c39a6 in GarbageCollectSharedAssets(bool, bool)
#13 0x007fd7a54dc551 in UnloadUnusedAssetsOperation::IntegrateMainThread()
#14 0x007fd7a54db5ec in PreloadManager::UpdatePreloadingSingleStep(PreloadManager::UpdatePreloadingFlags, int)
#15 0x007fd7a54dc03a in PreloadManager::UpdatePreloading()
#16 0x007fd7a54c5957 in ExecutePlayerLoop(NativePlayerLoopSystem*)
#17 0x007fd7a54c5912 in ExecutePlayerLoop(NativePlayerLoopSystem*)
#18 0x007fd7a54c5ebe in PlayerLoop()
#19 0x007fd7a56a3bab in PlayerMain(int, char**)
#20 0x007fd7a3c42d85 in __libc_start_main
#21 0x00564b10d8e029 in _start

**Comment by ToeBeeBee** [comment]
I have also been having the same issue, and would love more info on how to fix this. In the meantime have been using this  [#3468 ](https://github.com/GameServerManagers/LinuxGSM/discussions/3468#discussioncomment-1260080) to keep the server online after crashes

> Could not allocate 67108864 bytes
Caught fatal signal - signo:6 code:-6 errno:0 addr:0x3e800004901
Obtained 22 stack frames.
#0  0x007f235ad6f730 in funlockfile
#1  0x007f235abd48eb in gsignal
#2  0x007f235abbf535 in abort
#3  0x007f235a629ae8 in mono_dl_fallback_unregister
#4  0x007f235a639e94 in monoeg_g_logv
#5  0x007f235a639f27 in monoeg_g_log
#6  0x007f235a639b3a in monoeg_realloc
#7  0x007f235a63cb54 in monoeg_g_string_erase
#8  0x007f235a63cb92 in monoeg_g_ptr_array_sized_new
#9  0x007f235a61fb3e in mono_monitor_enter_v4
#10 0x007f235a61fd04 in mono_unity_liveness_allocate_struct
#11 0x007f235a620a2f in mono_unity_liveness_calculation_begin
#12 0x007f235bc5e9a6 in GarbageCollectSharedAssets(bool, bool)
#13 0x007f235bc77551 in UnloadUnusedAssetsOperation::IntegrateMainThread()
#14 0x007f235bc765ec in PreloadManager::UpdatePreloadingSingleStep(PreloadManager::UpdatePreloadingFlags, int)
#15 0x007f235bc7703a in PreloadManager::UpdatePreloading()
#16 0x007f235bc60957 in ExecutePlayerLoop(NativePlayerLoopSystem*)
#17 0x007f235bc60912 in ExecutePlayerLoop(NativePlayerLoopSystem*)
#18 0x007f235bc60ebe in PlayerLoop()
#19 0x007f235be3ebab in PlayerMain(int, char**)
#20 0x007f235abc109b in __libc_start_main
#21 0x00562ca1a5a029 in _start

**Reply by ToeBeeBee** [reply]
Upgraded my VPS to 8GB of RAM and it's stopped the crashing so far. I hope it's just a memory mismanagement issue and not just the fact that the game will use more memory from now onward.

**Comment by Ksando** [comment]
I'd like to note. I needed to restart the server at the OS level. After that, the valheim server has now been running without interruption for 32 hours, although before that it could not work even an hour. Of course, this is not a solution to the problem, but as a temporary solution, it will do.

**Reply by ToeBeeBee** [reply]
Hm, I've attempted OS updates and reboot but as soon as more than 1 or 2 people play it will go down after an hour or so
